core/src/main/java/de/danoeh/antennapod/core/storage/DatabaseExporter.java

```diff
@@ -53,7 +54,18 @@ public static void exportToStream(FileOutputStream outFileStream, Context contex
             if (currentDB.exists()) {
                 src = new FileInputStream(currentDB).getChannel();
                 dst = outFileStream.getChannel();
-                dst.transferFrom(src, 0, src.size());
+                long srcSize = src.size();
+                long oldDstSize = dst.size();
+                dst.transferFrom(src, 0, srcSize);
+
+                long newDstSize = dst.size();
+                long expected = srcSize + oldDstSize;
```

Comment:
The old file should be fully overwritten. Is the old size really needed?

Reply:
True, it's not.